randovania/resolver/bootstrap.py

```diff
@@ -250,29 +251,57 @@ def all_preplaced_pickup_locations(
 
         return locations
 
-    def pre_place_pickups(
+    def pre_place_pickups_weighted(
         self,
         rng: Random,
-        locations: list[PickupNode],
+        locations: dict[PickupNode, float],
         pool_results: PoolResults,
         item_category: str,
         game: RandovaniaGame,
     ) -> None:
         pre_placed_indices = list(pool_results.assignment.keys())
-        reduced_locations = [loc for loc in locations if loc.pickup_index not in pre_placed_indices]
+        reduced_locations = [loc for loc in locations.keys() if loc.pickup_index not in pre_placed_indices]
 
-        rng.shuffle(reduced_locations)
+        # working_locations is a copy of locations with only the keys in reduced locations
+        working_locations = {}
+        for loc in reduced_locations:
+            working_locations[loc] = locations[loc]
```

Comment:
`reduced_locations` is a list with only the keys and `working_locations` contains the same keys but with the weighting again, then this block is just:
```python
reduced_locations = {loc: v for loc, v in locations.items() if loc.pickup_index not in pre_placed_indices}
```